### PR TITLE
feat(oxc_language_server):  add diagnostic code with link

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
1. follow #6479 
2. Adding related doc link to lsp diagnostic code.
## Preview
image

**after click**
image

The UI looks a little wired, due to https://github.com/oxc-project/oxc/blob/aba9ea3f1d68c8b32e2b8aba7d6038737f1652eb/crates/oxc_diagnostics/src/lib.rs#L113-L120, 

I suggest change the code above to 
```rs
format!("{scope}/{number}")
```

### CODE REVIEW

File: crates/oxc_language_server/src/linter.rs [hunk 1]
```diff
@@ -107,7 +110,13 @@ impl ErrorWithPosition {
                 ret_range
             },
         );
-        let source = self.miette_err.code().map_or("oxc".to_string(), |code| format!("oxc:{code}"));
+        let code = self.miette_err.code().map(|item| item.to_string());
+        let code_description = code.as_ref().and_then(|code| {
```
Comment (DonIsaac) [inline crates/oxc_language_server/src/linter.rs:114]: If you need a doc link, maybe we add it to OxcDoagnostic directly?

**Review by DonIsaac — APPROVED** [review]

File: crates/oxc_language_server/src/linter.rs (hunk 1 above)
Comment (IWANABETHATGUY) [reply crates/oxc_language_server/src/linter.rs:114]: Yeah, that should be the most perfect solution. but I'm not sure if adding a new field hurt linter performance.

Comment (DonIsaac) [reply crates/oxc_language_server/src/linter.rs:114]: CPU tends to be cheaper than memory. It all depends on this: when we provide diagnostics as JSON, do we want to make those links available? This affects use cases in external crates/applications

Comment (DonIsaac) [reply crates/oxc_language_server/src/linter.rs:114]: Also: please consider this comment non blocking. This PR looks good to me.

Comment (IWANABETHATGUY) [reply crates/oxc_language_server/src/linter.rs:114]: Ok, let's polish in the future.

**Comment by IWANABETHATGUY** [comment]
### Merge activity

* **Oct 13, 1:01 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**